From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Toffikk <slodkitofik@gmail.com>
Date: Thu, 17 Jun 2021 14:44:11 +0200
Subject: [PATCH] (Sugarcane) Fix Javadoc generation

Original code by SugarcaneMC, licensed under GNU General Public License v3.0
You can find the original code on https://github.com/SugarcaneMC/Sugarcane

diff --git a/build.gradle.kts b/build.gradle.kts
index 02f80c9a34a23c2285a1e05b41f53b4bafc28c5a..e9b99787cf31bc6e40e5f55dc79ed158b07f0fb6 100644
--- a/build.gradle.kts
+++ b/build.gradle.kts
@@ -91,6 +91,8 @@ tasks.jar {
 
 tasks.withType<Javadoc>().configureEach {
     (options as CoreJavadocOptions).addStringOption("sourcepath", apiAndDocs.resolvedConfiguration.files.joinToString(separator = File.pathSeparator, transform = File::getPath))
+	(options as StandardJavadocDocletOptions).encoding = "UTF-8"
+	(options as StandardJavadocDocletOptions).addBooleanOption("html5", true)
     (options as StandardJavadocDocletOptions).links(
         "https://guava.dev/releases/21.0/api/docs/",
         "https://javadoc.io/doc/org.yaml/snakeyaml/1.28/",
